doc: SUIT recovery and customize DFU update (#19686)
nrfconnect/sdk-nrf: doc/nrf/app_dev/device_guides/nrf54h/ug_nrf54h20_suit_recovery.rst

If a failure during a boot process occurred, the Secure Domain sets the recovery flag and reboots the device.
Upon each boot the Secure Domain checks if the recovery flag is set.
If it is set, the device enters recovery mode.

> If it isn't, booting proceeds normally by running the root manifest.

Reviewer: ```suggestion
If it is not, the boot process continues normally by executing the root manifest.
```

After entering the recovery mode it is verified if MPI configuration for the APP_RECOVERY is present.
If it is, the APP_RECOVERY manifest is processed.

> If no MPI configuration is found the Secure Domain performs an attempt to process the normal manufacturer root manifest.

Reviewer: ```suggestion
If no MPI configuration is found, the Secure Domain attempts to process the normal manufacturer root manifest.
```

   This will cause the recovery firmware to be built automatically as part of the main application build.

#. :ref:`Program the main application firmware to the device <programming>`.

>    This will automatically program both the main application and the recovery firmware to the device.

Reviewer: ```suggestion
   This automatically programs both the main application and the recovery firmware onto the device.
```
